Feature/project status

tnagorra opened the pull request:
## Changes

- Update project status
  - Added status for draft projects
  - Added status for paused projects
- Add project processing status
- Update project queries and mutations
  - Add fields on project create
  - Add fields on project update
  - Add mutation for ready project update
  - Add validation for status transition
  - Add limitation on project update
- Rename functions and methods to be more generic

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests

tnagorra committed:
Update project status

- Added status for draft projects
- Added status for paused projects

tnagorra committed:
Rename functions and methods to be more generic

tnagorra committed:
Change project create and update mutation

- Add fields on project create
- Add response fields on project create
- Add fields on project update
- Add response fields on project update
- Add mutation for ready project update
- Add status transition validation
- Add limitation on project update

tnagorra committed:
amend! Change project create and update mutation

Change project create and update mutation

- Add fields on project create
- Add response fields on project create
- Add fields on project update
- Add response fields on project update
- Add mutation for ready project update
- Add status transition validation
- Add limitation on project update

tnagorra committed:
Update tests for project creation and update

- Update state after processing project

tnagorra committed:
Rename "ReadyProject" to "ProcessedProject"

- Fix async issue with update_processed_project
- Update failed state if process_project_task gets an exception
- Add filters for projects query

tnagorra committed:
Implement processing status for project

tnagorra committed:
Add test for processed project update

tnagorra committed:
Add schema and migration

tnagorra committed:
amend! Add schema and migration

Add schema and migration

thenav56 commented on apps/project/models.py:
```diff
@@ -39,12 +39,62 @@ def get_display(cls, value: typing.Self | int) -> str:
+
+
+class ProjectProcessingStatusEnum(models.IntegerChoices):
+    WAITING = 10, "Waiting"
```
Nullable field

thenav56 commented on apps/project/models.py:
```diff
@@ -194,7 +250,13 @@ class Project(UserResource):
+        return ProjectTypeEnum(self.project_type)
+
+    def update_status(self, status: ProjectStatusEnum, commit: bool = True):
+        # TODO(tnagorra): Add a validation here
```
Validation not required

tnagorra committed:
Make processing_status non-mandatory

- Remove "WAITING" value from ProjectProcessingStatusEnum
- Use update_status and update_processing_status
